imageregistry/v1/types.go

```diff
@@ -160,6 +160,19 @@ type ImageRegistryConfigStorageS3CloudFront struct {
 type ImageRegistryConfigStorageEmptyDir struct {
 }
 
+// S3TrustedCASource references a config map with a CA certificate bundle in
+// the "openshift-config" namespace. The key for the bundle in the
+// config map is "ca-bundle.crt".
+type S3TrustedCASource struct {
+	// name is the metadata.name of the referenced config map.
```

Comment:
Thanks for adding the validations here, the validation pattern and requirements are useful for computers but these aren't easily interpretably to an end user, would be good to add some prose to explain the validations
Eg
```
// The name must consist solely of alphanumeric characters, hyphens (-) and periods (.).
// It has a maximum length of 253 characters.
```

Reply:
The name must be a valid config map name. No extra restrictions.

Reply:
This config map should be created by a user.

Reply:
Yeah that makes sense, my point is we should tell the user this in the godoc, as this becomes generated documentation. I think the comments I suggested satisfy my requirement on this

Reply:
If you want to know more about why or how we write the godocs, check out the API conventions section, https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#write-user-readable-documentation-in-godoc

Reply:
Yeah, I know that. I find these additional comments slightly confusing as they create feeling that not every config map can be used there, which isn't true.

Reply:
That is a fair point, do you think it would be clear if we added another line or are you a hard no on adding any extra?
```
// This field must adhere to standard config map naming restrictions.
// The name must consist solely of alphanumeric characters, hyphens (-) and periods (.).
// It has a maximum length of 253 characters.
```